src/main/java/appeng/client/me/FluidRepo.java

```diff
@@ -0,0 +1,253 @@
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.client.me;
+
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+import javax.annotation.Nonnull;
+
+import appeng.api.AEApi;
+import appeng.api.config.Settings;
+import appeng.api.config.SortOrder;
+import appeng.api.config.ViewItems;
+import appeng.api.config.YesNo;
+import appeng.api.storage.channels.IFluidStorageChannel;
+import appeng.api.storage.data.IAEFluidStack;
+import appeng.api.storage.data.IItemList;
+import appeng.client.gui.widgets.IScrollSource;
+import appeng.client.gui.widgets.ISortSource;
+import appeng.core.AEConfig;
+import appeng.util.FluidSorters;
+import appeng.util.Platform;
+import appeng.util.prioritylist.IPartitionList;
+
+
+/**
+ * @author BrockWS
+ * @version rv6 - 22/05/2018
+ * @since rv6 22/05/2018
+ */
+public class FluidRepo
+{
+	private final IItemList<IAEFluidStack> list = AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ).createList();
+	private final ArrayList<IAEFluidStack> view = new ArrayList<>();
+	private final IScrollSource src;
+	private final ISortSource sortSrc;
+
+	private int rowSize = 9;
+
+	private String searchString = "";
+	private IPartitionList<IAEFluidStack> myPartitionList;
+	private String innerSearch = "";
+	private boolean hasPower;
+
+	public FluidRepo( final IScrollSource src, final ISortSource sortSrc )
+	{
+		this.src = src;
+		this.sortSrc = sortSrc;
+	}
+
+	public void updateView()
+	{
+		this.view.clear();
+
+		this.view.ensureCapacity( this.list.size() );
+
+		final Enum viewMode = this.sortSrc.getSortDisplay();
+		final Enum searchMode = AEConfig.instance().getConfigManager().getSetting( Settings.SEARCH_MODE );
+		final boolean needsZeroCopy = viewMode == ViewItems.CRAFTABLE;
+
+		this.innerSearch = this.searchString;
+		final boolean terminalSearchToolTips = AEConfig.instance().getConfigManager().getSetting( Settings.SEARCH_TOOLTIPS ) != YesNo.NO;
+
+		boolean searchMod = false;
+		if( this.innerSearch.startsWith( "@" ) )
+		{
+			searchMod = true;
+			this.innerSearch = this.innerSearch.substring( 1 );
+		}
+
+		Pattern m;
+		try
+		{
+			m = Pattern.compile( this.innerSearch.toLowerCase(), Pattern.CASE_INSENSITIVE );
+		}
+		catch( final Throwable ignore )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1181)